OBSDOCS-552:  sample service config for Cluster Observability Operator tech preview (#67608)
openshift/openshift-docs: modules/monitoring-creating-a-monitoringstack-object-for-cluster-observability-operator.adoc

// Module included in the following assemblies:
//
// monitoring/cluster-observability-operator/configuring-the-cluster-observability-operator-to-monitor-a-service.adoc

:_mod-docs-content-type: PROCEDURE
[id="creating-a-monitoringstack-object-for-cluster-observability-operator_{context}"]

> = Creating a MonitoringStack object for Cluster Observability Operator

Reviewer: ```suggestion
= Creating a MonitoringStack object for the Cluster Observability Operator
```
Maybe consider adding an attribute for the Operator name to save you from typing it out every time?

Author: Fixed.

I'll also add an attribute, and be thankful for it later.

> To be able to discover the sample `ServiceMonitor` resource you created, you must create and configure a `MonitoringStack` object.

Reviewer: ```suggestion
To be able to discover the sample `ServiceMonitor` object you created using the documentation on "Specifying how a service is monitored for Cluster Observability Operator", you must create and configure a `MonitoringStack` object.
```

I'd maybe add some explanation of "the sample `ServiceMonitor` resource you created" since it's not immediately clear out of content what that would be. Can users only use this procedure with that `ServiceMonitor` object, or can it be any `ServiceMonitor` object?

Also, I'd use "object" here as well since you use  `MonitoringStack` object, or else use "resource" throughout.

Author: Changed per your suggestions.

> This procedure shows how to create a `MonitoringStack` object to discover the sample `ServiceMonitor` resource for a sample service named `prometheus-coo-example-app` in the `ns1-coo` namespace.

Reviewer: Tbh I'd maybe actually rewrite the intro a bit, and instead of the two paragraphs have something like:

"You can create a `MonitoringStack` object to discover the sample `ServiceMonitor` object.  The `ServiceMonitor` object is for a sample service named `prometheus-coo-example-app` in the `ns1-coo` namespace."

That way it:
- breaks up a long sentence
- removes ambiguity referring to a different (earlier) procedure / `ServiceMonitor` resource
- removes self-referencing from the intro

Just a suggestion though 🙂

Author: Good suggestion. Done

.Prerequisites

> * You have access to the cluster as a user with the `cluster-admin` cluster role or as a user with administrative permissions for the namespace.

Author: I don't think a comma is needed here. Leaving as-is to be consistent with the way it is throughout the monitoring docs.

> * You have installed the Cluster Observability Operator.

Reviewer: recommend adding an attribute for the Operator name

Author: Agree. Will do.

* You have deployed the `prometheus-coo-example-app` sample service in the `ns1-coo` project.

> * You have configured a custom resource to specify how the service is to be monitored.
> 

Reviewer: ```suggestion
* You have created a `ServiceMonitor` object.
```
Maybe keep this consistent with the mention of this object earlier if it has the same meaning here?

Author: Good point. Done.

.Procedure

. Create a YAML file for the `MonitoringStack` object configuration. For this example, name the file `example-coo-monitoring-stack.yaml`.

. Add the following `MonitoringStack` object configuration details:

> +
> [source,yaml]

Reviewer: ```suggestion
+
.Example `MonitoringStack` object
[source,yaml]
```

Author: Added.

----
apiVersion: monitoring.rhobs/v1alpha1
kind: MonitoringStack
metadata:
  name: example-coo-monitoring-stack
  namespace: ns1-coo
spec:
  logLevel: debug
  retention: 1d
  resourceSelector:
    matchLabels:
      k8s-app: prometheus-coo-example-monitor
----

> . Apply the configuration to the cluster:

Reviewer: ```suggestion
. Apply the `MonitoringStack` object by running the following command:
```

Author: Indeed! Fixed.

+
[source,terminal]
----
$ oc apply -f example-coo-monitoring-stack.yaml
----

> . Verify that the `MonitoringStack` object is available:

Reviewer: ```suggestion
. Verify that the `MonitoringStack` object is available, by running the following command and inspecting the output:
```

Author: Added everything but the comma, which I don't think is needed.

// Module included in the following assemblies:
//
// monitoring/cluster-observability-operator/configuring-the-cluster-observability-operator-to-monitor-a-service.adoc

:_mod-docs-content-type: PROCEDURE
[id="creating-a-monitoringstack-object-for-cluster-observability-operator_{context}"]
= Creating a MonitoringStack object for the Cluster Observability Operator

> To be able to discover the sample `ServiceMonitor` object you created by using the documentation in "Specifying how a service is monitored for Cluster Observability Operator", you must create and configure a `MonitoringStack` object.

Reviewer 2: in "Specifying how a service is monitored for Cluster Observability Operator"
should be changed as 
in the "Specifying how a service is monitored for Cluster Observability Operator" section
better provide a link

Author: It's not possible to add a link here because of restrictions in the OCP docs repo.

You can create a `MonitoringStack` object to discover the sample `ServiceMonitor` object. 

> The `ServiceMonitor` object is for a sample service named `prometheus-coo-example-app` in the `ns1-coo` namespace."

Reviewer: ```suggestion
The `ServiceMonitor` object is for a sample service named `prometheus-coo-example-app` in the `ns1-coo` namespace.
```

Reviewer: Is there a way to clarify "is for" here? Maybe we could say something like "The `ServiceMonitor` object monitors the sample service" or be more specific about what it does? Not a blocker or anything, just curious.

Author: - Fixed the typo. 
- Excellent idea. Updated per your suggested wording.

> The `ServiceMonitor` object monitors the sample service named `prometheus-coo-example-app` in the `ns1-coo` namespace.

Reviewer 2: @abrennan89 You are wrong here for`ServiceMonitor` object is config data and can't monitor anything. @bburt-rh You may change line 11 and 12 as the following, line 50 has given detail how `ServiceMonitor` specify services to be monitored.
 You can create a `MonitoringStack` object to discover the target service to be monitored by using`ServiceMonitor` object and scrape data from the target.

Author: @lihongyan1 
I made changes that I think should address this issue. Can you please review again? Thanks.
